### ISSUE TITLE
App crushes when I try to add networkSaga to rootSaga

### ISSUE THREAD

**Issue by olegklimakov** [rgommezz/react-native-offline#251]
### Current Behavior
```
import { all, fork, takeEvery, call } from 'redux-saga/effects';
import { networkSaga } from 'react-native-offline';

export default function* rootSaga () {
    yield all([
        fork(counters),
        fork(networkSaga, { pingInterval: 20000 }),
    ]);
}
```
// store.ts

```
import { applyMiddleware, combineReducers, createStore } from 'redux';
import auth from './reducers/auth.reducer';
import { createNetworkMiddleware, reducer as network } from 'react-native-offline';
import createSagaMiddleware from 'redux-saga';
import rootSaga from './sagas/root-saga';
import counters  from './counters/counters.reducer';
// Middleware: Redux Saga
const sagaMiddleware = createSagaMiddleware();
const networkMiddleware = createNetworkMiddleware({
    queueReleaseThrottle: 200,
});

const rootReducer = combineReducers({
    auth,
    counters,
    network
});

const store = createStore(rootReducer, applyMiddleware(networkMiddleware, sagaMiddleware));
sagaMiddleware.run(rootSaga);
export default store;
```

Also have error on this line fork(networkSaga, { pingInterval: 20000 }),
```TS2769: No overload matches this call.   The last overload gave the following error.     Argument of type '(args?: ConnectivityArgs) => Generator<SimpleEffect<"FORK", ForkEffectDescriptor<any>>, void, unknown>' is not assignable to parameter of type '{ context: unknown; fn: (this: unknown, ...args: any[]) => any; }'.       Type '(args?: ConnectivityArgs) => Generator<SimpleEffect<"FORK", ForkEffectDescriptor<any>>, void, unknown>' is missing the following properties from type '{ context: unknown; fn: (this: unknown, ...args: any[]) => any; }': context, fn  effects.d.ts(678, 17): The last overload is declared here.  ```
When I try to run application nothing happened  - expo crushed and goes to home screen

### Expected Behavior

The app shouldn't crush

### How to reproduce

https://gist.github.com/olegklimakov/95042b9fb54f7ba2812359db05fed5ec

### Your Environment
"@react-native-community/masked-view": "^0.1.6",
    "@react-native-community/netinfo": "4.6.0",
    "@react-navigation/drawer": "^5.0.0",
    "@react-navigation/native": "^5.0.0",
    "@react-navigation/stack": "^5.0.0",
    "@types/react-redux": "^7.1.7",
    "@types/redux-thunk": "^2.1.0",
    "dayjs": "^1.8.20",
    "expo": "~36.0.0",
    "expo-file-system": "~8.0.0",
    "expo-image-picker": "~8.0.1",
    "expo-sqlite": "^8.0.0",
    "g": "^2.0.1",
    "ky": "^0.17.0",
    "react": "~16.9.0",
    "react-dom": "~16.9.0",
    "react-native": "https://github.com/expo/react-native/archive/sdk-36.0.0.tar.gz",
    "react-native-gesture-handler": "~1.5.0",
    "react-native-htmlview": "^0.15.0",
    "react-native-offline": "^5.4.0",
    "react-native-reanimated": "~1.4.0",
    "react-native-safe-area-context": "^0.6.0",
    "react-native-screens": "2.0.0-alpha.12",
    "react-native-svg": "9.13.6",
    "react-native-vector-icons": "^6.6.0",
    "react-native-web": "~0.11.7",
    "react-navigation": "^4.0.10",
    "react-navigation-header-buttons": "^3.0.4",
    "react-navigation-material-bottom-tabs": "^2.1.5",
    "react-navigation-tabs": "^2.7.0",
    "react-redux": "^7.1.3",
    "redux": "^4.0.5",
    "redux-saga": "^1.1.3"

**Comment by olegklimakov** [comment]
Also, example app falls on IOS device and IOS emulator

**Comment by mariusz360** [comment]
My app crashes when I wrap my App component with ReduxNetworkProvider. I think it may be related to this issue.
```
<ReduxNetworkProvider>
  <App />
</ReduxNetworkProvider>
```

**Comment by brunomazzardo** [comment]
Having the same problem as @mariusz360, found any solution?

**Comment by flexbox** [comment]
I am facing the same issue 😢 
```
    "expo": "^36.0.0",
    "react-native-offline": "^5.4.0",
```
I followed the README insttructions and adding the `<NetworkProvider>` crash the app.

Thanks @mariusz360 for the suggestion
I fixed the version to `"react-native-offline": "5.2.0",` and it works now 🚀

**Comment by rgommezz** [comment]
Hey guys,

Could you please try the version `5.6.0` and let me know?

I've updated the documentation and relaxed the dependencies, to make it compatible with both RN >= 60 and Expo SDK >=35.

**Comment by flexbox** [comment]
Hey @rgommezz 
Thank you for your work on this 🙌

I tried to:
1. bump the dependencies
```
"react-native-offline": "^5.6.0",
```
2. clear the cache
```
watchman watch-del-all && rm -rf node_modules yarn.lock && yarn install
```

I still have this error

![image](https://user-images.githubusercontent.com/360936/78878023-03ed1f00-7a52-11ea-816c-5e30abbd7778.png)

Same on snack 
https://snack.expo.io/@flexbox/bad-pastry

![image](https://user-images.githubusercontent.com/360936/78879023-9e9a2d80-7a53-11ea-8aa9-ae1cfd907a9f.png)

**Comment by rgommezz** [comment]
Hi @flexbox, thanks to you to test it out so quickly.

I think I've figured out the problem.

Please do try `5.6.1` and lemme know

**Comment by flexbox** [comment]
I got the kind of the same error.
- Crash on iOS without mounting the app
- on Android
![image](https://user-images.githubusercontent.com/360936/78922639-29504c00-7a97-11ea-8f2b-02fe0baeaced.png)

I try to bump to the expo SDK 37 with

```
npm i -g expo-cli
expo upgrade
yarn cache:clear
```

It's working on Android and iOS 🎉

I think you can add a mention on the README because with the expo SDK 36 I am not sure about the version of `react-native-netinfo` shipped

**Comment by YoshiYo** [comment]
Heya !

I had the same problem.

My expo literally crashed, I had such a hard time to find this issue in the repo to fix my problem. No stacktrace or what was provided.. :'(

So, I was in `^5.2.0` and I try `5.2.0` and it works well.
I tried the `5.6.1` but I crashed too.

I think it's not a problem on expo (or maybe they did some changes in SDK 36), but I tried to upgrade in SDK 37 before going on this issue and it doesn't fix it. 

But I didn't do exactly your procedure @flexbox, so I will try it in the next days and comeback here to give you some insights too. What is your actual version of `react-native-offline` now ?
